MyFace/Controllers/UserListController.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Web.Mvc;
+using MyFace.DataAccess;
+using MyFace.Helpers;
+using MyFace.Middleware;
+using MyFace.Models.ViewModels;
+
+namespace MyFace.Controllers
+{
+    [BasicAuthentication]
+    public class UserListController : Controller
+    {
+        private readonly IUserRepository userRepository;
+
+        public UserListController(IUserRepository userRepository)
+        {
+            this.userRepository = userRepository;
+        }
+
+        public ActionResult Index()
+        {
+            var username = AuthenticationHelper.ExtractUserNameAndPassword(Request).Username;
```

Comment:
This is slightly sad because we already use this method in the Basic authentication.  However I don't think there is any easy way to get this information into the controller from the Action filter.